pkg/probe/http/http.go

```diff
@@ -101,6 +101,14 @@ func DoHTTPProbe(url *url.URL, headers http.Header, client GetHTTPInterface) (pr
 		v := version.Get()
 		headers.Set("User-Agent", fmt.Sprintf("kube-probe/%s.%s", v.Major, v.Minor))
 	}
+	if _, ok := headers["Accept"]; !ok {
+		// Accept header was not defined. accept all
+		headers.Set("Accept", "*/*")
+	}
+	if headers.Get("Accept") == "" {
+		// Accept header was overridden but is empty. removing
+		headers.Del("Accept")
+	}
 	req.Header = headers
 	if headers.Get("Host") != "" {
 		req.Host = headers.Get("Host")
```

Comment:
unrelated, I'm surprised that Host is not being removed from the collection here.

Reply:
The Host header is required for HTTP 1.1 from what I read. So this should be OK

Reply:
setting `req.Host` must be enough =). It looks like `Host` header in the headers collection will be ignored (https://github.com/golang/go/blob/cad6d1fef5147d31e94ee83934c8609d3ad150b7/src/net/http/request.go#L92). I wasn't sure about whether the header in the collection will be ignored, thus the comment. From code understanding, removing it from the collection would be logical for me here.

Reply:
again, totally irrelevant to this PR.

Reply:
Got it, you mean the condition can be removed because the Headers will always have a Host. Yeah, if you prefer I can remove the condition here or let it for a new PR.

Reply:
no, let's not touch this to minimize changes